src/webgl/p5.Framebuffer.js

```diff
@@ -0,0 +1,552 @@
+    if (!depthTexture) {
+      throw new Error('Unable to create depth texture');
+    }
+    const depthFormat = this._glDepthFormat();
+    gl.bindTexture(gl.TEXTURE_2D, depthTexture);
+    gl.texImage2D(
+      gl.TEXTURE_2D,
+      0,
+      depthFormat.internalFormat,
+      this.width * this.density,
+      this.height * this.density,
+      0,
+      depthFormat.format,
+      depthFormat.type,
+      null
+    );
+
+    gl.bindFramebuffer(gl.FRAMEBUFFER, this.framebuffer);
+    gl.framebufferTexture2D(
+      gl.FRAMEBUFFER,
+      gl.COLOR_ATTACHMENT0,
+      gl.TEXTURE_2D,
+      colorTexture,
+      0
+    );
+    gl.framebufferTexture2D(
+      gl.FRAMEBUFFER,
+      gl.DEPTH_ATTACHMENT,
+      gl.TEXTURE_2D,
+      depthTexture,
+      0
+    );
+
+    // Create separate framebuffer for antialiasing
+    if (this.antialias) {
+      this.colorRenderbuffer = gl.createRenderbuffer();
+      gl.bindRenderbuffer(gl.RENDERBUFFER, this.colorRenderbuffer);
+      gl.renderbufferStorageMultisample(
+        gl.RENDERBUFFER,
+        Math.min(4, gl.getParameter(gl.MAX_SAMPLES)),
+        colorFormat.internalFormat,
+        this.width * this.density,
+        this.height * this.density
+      );
+
+      this.depthRenderbuffer = gl.createRenderbuffer();
+      gl.bindRenderbuffer(gl.RENDERBUFFER, this.depthRenderbuffer);
+      gl.renderbufferStorageMultisample(
+        gl.RENDERBUFFER,
+        Math.min(4, gl.getParameter(gl.MAX_SAMPLES)),
+        depthFormat.internalFormat,
+        this.width * this.density,
+        this.height * this.density
+      );
+
+      gl.bindFramebuffer(gl.FRAMEBUFFER, this.aaFramebuffer);
+      gl.framebufferRenderbuffer(
+        gl.FRAMEBUFFER,
+        gl.COLOR_ATTACHMENT0,
+        gl.RENDERBUFFER,
+        this.colorRenderbuffer
+      );
+      gl.framebufferRenderbuffer(
+        gl.FRAMEBUFFER,
+        gl.DEPTH_ATTACHMENT,
+        gl.RENDERBUFFER,
+        this.depthRenderbuffer
+      );
+    }
+
+    this.depthTexture = depthTexture;
+    this.colorTexture = colorTexture;
+
+    this.depth = new FramebufferTexture(this, 'depthTexture');
+    const depthP5Texture = new p5.Texture(
+      this.target._renderer,
+      this.depth,
+      {
+        minFilter: gl.NEAREST,
+        magFilter: gl.NEAREST
+      }
+    );
+    this.target._renderer.textures.set(this.depth, depthP5Texture);
+
+    this.color = new FramebufferTexture(this, 'colorTexture');
+    const colorP5Texture = new p5.Texture(
+      this.target._renderer,
+      this.color,
+      {
+        glMinFilter: gl.LINEAR,
+        glMagFilter: gl.LINEAR
+      }
+    );
+    this.target._renderer.textures.set(this.color, colorP5Texture);
+
+    gl.bindTexture(gl.TEXTURE_2D, prevBoundTexture);
+    gl.bindFramebuffer(gl.FRAMEBUFFER, prevBoundFramebuffer);
+  }
+
+  _glColorFormat() {
+    let type, format, internalFormat;
+    const gl = this.gl;
+
+    if (this.format === constants.FLOAT) {
+      type = gl.FLOAT;
```

Comment:
Do we need to check for webGL2 compatibility here too?
https://developer.mozilla.org/en-US/docs/Web/API/OES_texture_float

Reply:
I've got it set up so that `_checkIfFormatsAvailable()` will set the format to something else if the user asked for FLOAT but that isn't available, so that in the rest of the code we can just assume that we have access to the extensions and such that would be required.